dataset/src/main/scala/frameless/Job.scala

```diff
@@ -22,8 +22,17 @@ sealed abstract class Job[A](implicit sc: SparkContext) { self =>
       }
     }
   }
+
+  def map[B](fn: A => B): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run())
+  }
+
+  def flatMap[B](fn: A => Job[B]): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run()).run()
+  }
```

Comment:
What about making `Dataset` depend on cats? Then we can just get rid of `Job` and replace with `Eval`, which already has `Functor` and `Monad` (with law checks) and is stack-safe (unlike `Job`)